.gitlab-ci.yml

```diff
@@ -20,6 +20,17 @@ include:
     ref: latest
     file: 'front-end-publish-npm.yml'
 
+type-check:
+  retry: 2
+  stage: test
+  script:
+    - yarn type-check
+  tags:
+    - shell
+    - dev
+  except:
+    - tags
+
```

Comment:
I'm not attached to this name but we should get the name here correct (type-check, typescript, typescript-check, etc), and I can file a devops ticket and give them this yml file partial to put in gitlab includes

Reply:
I'm onboard with a more generic name (no "typescript"). This step is about static type checks. I like it.